README.md
<!--
SPDX-FileCopyrightText: © 2020 Jean-Pierre De Jesus DIAZ <me@jeandudey.tech>
SPDX-License-Identifier: MIT OR Apache-2.0
-->

# `tokio-udev`

> Non-blocking/asynchronous mio & tokio support for udev device events

# [Documentation](https://docs.rs/tokio-udev)

# Minimum Supported Rust Version

The crates are guaranteed to compile on stable Rust 1.56. It _might_ compile
with older versions but that may change in any new patch release.

# License

Licensed under either of

- Apache License, Version 2.0 ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
- MIT License ([LICENSE-MIT](LICENSE-MIT) or http://opensource.org/licenses/MIT)

at your option.

# Contribution

Unless you explicitly state otherwise, any contribution intentionally submitted
for inclusion in the work by you, as defined in the Apache-2.0 license, shall be
dual licensed as above, without any additional terms or conditions.


CHANGELOG.md
<!--
SPDX-FileCopyrightText: © 2020 Jean-Pierre De Jesus DIAZ <me@jeandudey.tech>
SPDX-License-Identifier: MIT OR Apache-2.0
-->

# [unreleased]

# 0.9.1

- tokio-udev: update doc comment with new version.

# 0.9.0

- tokio-udev: update udev to 0.7.
- tokio-udev: set MSRV as 1.56.
- tokio-udev: fix bug where some events would not be retrieved (#17).
- tree-wide: use REUSE to identify copyright in each file.

# 0.8.0

- tokio-udev: by mistake the version released contains the same code as 0.7.0, no version
was tagged.

# 0.7.0

- tokio-udev: remove dependency to `mio-udev` and use the provided
mio implementation of the `udev` crate.
- tokio-udev: fix Send/Sync soundness (see #13). As a result, the Send/Sync implementations
are removed to avoid double-free/undefined behaviour.
- tokio-udev: rename `MonitorSocket` to `AsyncMonitorSocket` to keep the existing type from
udev.

# 0.6.0

- tokio-udev: use tokio `0.1`.

# 0.5.0

- mio-udev,tokio-udev: update udev to 0.5 [#12](https://github.com/jeandudey/tokio-udev/pull/12).

# 0.4.0

- tokio-udev: Update tokio to `>=0.3.2` [#9](https://github.com/jeandudey/tokio-udev/pull/9).
- tokio-udev: Update mio to `0.7` [#9](https://github.com/jeandudey/tokio-udev/pull/9).
- tokio-udev: Update mio-udev to `0.4` [#11](https://github.com/jeandudey/tokio-udev/pull/11).
- mio-udev: Update mio to `0.7`.

# 0.3.0

- .github: Add Rust actions ([#5](https://github.com/jeandudey/tokio-udev/pull/5))
- {mio,tokio}-udev: format code with rustfmt ([#4](https://github.com/jeandudey/tokio-udev/pull/4))
  - mio-udev: simplify if-expression
- Add missing documention ([#7](https://github.com/jeandudey/tokio-udev/pull/7))
  - CHANGELOG: add basic changelog
  - README: add README.md
  - {mio,tokio}-udev: update crate descriptions
  - {mio,tokio}-udev: apply copyright
  - Add LICENSE-MIT and LICENSE-APACHE
- mio-udev: update udev to 0.4 [#8](https://github.com/jeandudey/tokio-udev/pull/8)

# 0.2.0

- Update to tokio 0.2 ([#3](https://github.com/jeandudey/tokio-udev/pull/3))
 - Expose udev::Enumerator for public use
 - Migrate to Rust 2018
 - Update to tokio 0.2

# 0.1.0

- Initial Release


guix.scm
;;; SPDX-FileCopyrightText: © 2023 Jean-Pierre De Jesus DIAZ <me@jeandudey.tech>
;;; SPDX-License-Identifier: GPL-3.0-or-later

(use-modules (gnu packages)
             (gnu packages crates-io)
             (gnu packages linux)
             (gnu packages pkg-config)
             (guix build-system cargo)
             (guix download)
             (guix gexp)
             (guix git-download)
             ((guix licenses) #:prefix license:)
             (guix packages))

;;; Commentary:
;;
;; To build and install, run:
;;
;;   guix package -f guix.scm
;;
;; To build it, but not install it, run:
;;
;;   guix build -f guix.scm
;;
;; To use as the basis for a development environment, run:
;;
;;   guix environment -l guix.scm
;;
;;; Code:

(define-public rust-libudev-sys-0.1
  (package
    (name "rust-libudev-sys")
    (version "0.1.4")
    (source (origin
              (method url-fetch)
              (uri (crate-uri "libudev-sys" version))
              (file-name (string-append name "-" version ".tar.gz"))
              (sha256
                (base32
                  "09236fdzlx9l0dlrsc6xx21v5x8flpfm3d5rjq9jr5ivlas6k11w"))))
    (build-system cargo-build-system)
    (arguments
     `(#:cargo-inputs
       (("rust-libc" ,rust-libc-0.2)
        ("rust-pkg-config" ,rust-pkg-config-0.3))))
    (native-inputs (list pkg-config))
    (inputs (list eudev))
    (home-page "https://github.com/dcuddeback/libudev-sys")
    (synopsis "FFI bindings to libudev")
    (description "FFI bindings to libudev")
    (license license:expat)))

(define-public rust-udev-0.7
  (package
    (name "rust-udev")
    (version "0.7.0")
    (source (origin
              (method url-fetch)
              (uri (crate-uri "udev" version))
              (file-name (string-append name "-" version ".tar.gz"))
              (sha256
                (base32
                  "06hr927z0fdn7ay0p817b9x19i5fagmpmvz95yhl4d1pf3bbpgaf"))))
    (build-system cargo-build-system)
    (arguments
      `(#:cargo-inputs
        (("rust-libc" ,rust-libc-0.2)
         ("rust-libudev-sys" ,rust-libudev-sys-0.1)
         ("rust-mio" ,rust-mio-0.8)
         ("rust-mio" ,rust-mio-0.7)
         ("rust-mio" ,rust-mio-0.6)
         ("rust-pkg-config" ,rust-pkg-config-0.3))))
    (native-inputs (list pkg-config))
    (inputs (list eudev))
    (home-page "https://github.com/Smithay/udev-rs")
    (synopsis "libudev bindings for Rust")
    (description "libudev bindings for Rust")
    (license license:expat)))

(define-public rust-tokio-udev-0.9.0
  (package
    (name "tokio-udev")
    (version "0.9.0")
    (source (local-file "tokio-udev"
                        "rust-tokio-udev"
                        #:recursive? #t
                        #:select? (git-predicate "tokio-udev")))
    (build-system cargo-build-system)
    (arguments
      `(#:cargo-inputs
         (("rust-futures-core" ,rust-futures-core-0.3)
          ("rust-futures-util" ,rust-futures-util-0.3)
          ("rust-tokio" ,rust-tokio-1)
          ("rust-udev" ,rust-udev-0.7))))
    (native-inputs (list pkg-config))
    (inputs (list eudev))
    (home-page "https://github.com/jeandudey/tokio-udev")
    (synopsis "Monitor udev/eudev events asynchronously")
    (description "This Rust library provides abstractions to wait for
udev/eudev events asycnchronously using the Tokio runtime.")
    (license (list license:expat license:asl2.0))))

rust-tokio-udev-0.8
